pelican/__init__.py

```diff
@@ -19,15 +19,15 @@
 
 # pelican.log has to be the first pelican module to be loaded
 # because logging.setLoggerClass has to be called before logging.getLogger
```

Comment:
And if this is auto "fixed", it needs to be configured to not do that again...

Reply:
While it's good that you pointed this out, having a unit test would be even better.

I also added a function set_logger_class(), to call for the action explicitly instead of relying on a side effect.

The auto-fixing does not get rid of that function, so that is fixed.

Reply:
I don't consider this as "fixed". I should not need to "invent" interfaces to make a linter happy. This (`# noqa: I100`) was deliberate, so that the import order checks would not trigger.  If `ruff` auto fixer thinks that it can "remove noqa and reorder imports", then it is not safe to use.

```
from pelican.generators import (
    ArticlesGenerator,  # noqa: I100
```

Reply:
I'm not sure what to say.

Apparently we decided to use ruff.

Getting it to complain without fixing seems like a major flaw. No one will be able to auto-fix without this coming up again. So everyone will have to manually fix things until ruff stops complaining? That seems to put a lot of unneeded burden on developers and reviewers, slowing things down because of all the back and forth.

Relying on the side-effect of an import is more dangerous and hard to see than calling it out explicitly with a function, especially with no unit test.

So I don't understand where to go next.

@justinmayer ?

Reply:
> Relying on the side-effect of an import is more dangerous and hard to see than calling it out explicitly with a function, especially with no unit test.

This is beside the point. If auto fixing can result in behavior change (as in this example), it is simply unacceptable. It is *not safe* to use it.

> So I don't understand where to go next.

`# noqa` has a purpose. It means "I know this doesn't obey the rules but it is an exception. Ignore it.". I don't know why, but seems like `ruff` made up its own code for it ([I001](https://docs.astral.sh/ruff/rules/#isort-i)) instead of following what's been used in flake8. You can try changing the original to ` # noqa I100, I001` and see if that makes this bad behavior goes away.